### ISSUE TITLE
:bug: CollectionResources x Thumbnails: Alt text doesn't work

### ISSUE THREAD

**Issue by ShanaLMoore** [scientist-softserv/hykuup_knapsack#205]
# Summary

# Acceptance Criteria

- [ ] A user should be able to upload a banner, logo, and thumbnail to a collection, from the collection edit page 
- [ ] the banner, logo and thumbnail should be able to support alt text, if set

# Screenshots or Video

 collections can no longer upload a thumbnail. After creating a collection, a user used to be able to upload a thumbnail on edit.

<details>

![image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/880d0ab3-a883-43c8-b0bd-1c7fc3c842ed)

It should look like:

![image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/9649216f-6466-4b30-af8c-37b08446f689)

</details>

# Testing Instructions

- [ ] On [DEMO](https://demo.hyku-demo.notch8.cloud), Login as admin
- [ ] Create a New User Collection
- [ ] After clicking save, the edit page should have a "Branding" tab. Click it, and notice the "Thumbnail" section at the bottom of the form
- [ ] Upload an image and save. Note, Alt Text may look out of place - this is a known bug and considered a separate issue. Alt text is optional - enter it if you'd like. 
- [ ] Click Collections in the sidebar to view all the collections (index page)
- [ ] The thumbnail should appear and match the file you just uploaded
- [ ] If you entered alt text, you should see if when you inspect the thumbnail element of that page. ie: see eddie-baby-alt-text

<details> 

<img width="703" alt="image" src="https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/3f855b71-dd43-489e-8cba-c594414b6f2b">

</details>

### Results checklist 
**(verifying both branding AND default thumbnails, since they interact)**

- [ ] Setting Branding
    - [ ] Images and text can be set and will appear on collection's branding tab
    - [ ] Images and text can be removed and disappear from collection's branding tab
    - [ ] Branding shows appropriately on collection's public show page
    - [ ] Branding reverts to standard when removed from collection's branding tab
    - Note that banner alt-text appears in title, as background images cannot support alt-text.
- [ ] Setting Default thumbnails
    - [ ] Default thumbnails and alt text are saved and appear on Settings/Appearance default images tab
    - [ ] Default thumbnails and text can be removed and disappear from Settings/Appearance default images tab
    - [ ] Setting a default thumbnail or text reindexes all works &/or collections, and results in all collections or works showing new default thumbnails appropriately
    - [ ] Removing a collection's default thumbnail does not affect works
    - [ ] Removing a work's default thumbnail does not affect collections
- [ ] Thumbnails show appropriately on all views:
    - [ ] catalog search results view
    - [ ] dashboard index views
    - [ ] collection public show
    - [ ] collection dashboard show
    - [ ] homepage featured collections

# Notes

ref: 
-  https://github.com/samvera/hyrax/issues/5286

**Comment by ShanaLMoore** [comment]
https://samvera.slack.com/archives/CA8ANGLEL/p1714575345096909



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/ec8c6f54-96ae-48c3-8cfd-cae54fffd0c0)

**Comment by ShanaLMoore** [comment]
We will implement this the way that collection logos and banners got implemented. 

ref: https://github.com/samvera/hyku/pull/2204/files

**Comment by ShanaLMoore** [comment]
# QA Results: Needs Rework :X:

Uploading branding images are not loading in the UI. 

Example: https://demo.hyku-demo.notch8.cloud/dashboard/collections/5cd101a4-e698-4141-8c02-8976e14efe49?locale=en

Image Not Found error message. This works locally so I think it's a server issue. I tried adding a persistent volume in [this PR](https://github.com/samvera/hyku/pull/2219)

**Comment by ShanaLMoore** [comment]
UPDATED: the latest [deploy](https://demo.hyku-demo.notch8.cloud/dashboard/collections/5cd101a4-e698-4141-8c02-8976e14efe49/edit?locale=en#branding) renders the upload, but the thumbnails still don't show



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/456d50ff-b476-46cb-ad26-eb9588361601)



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/876cec64-63cc-4937-8926-a877a937ce5e)

**Comment by ShanaLMoore** [comment]
The issue seems to be whatever is responsible for indexing thumbnail_path_ss

staging data: 

```
 "thumbnail_path_ss"=>"/app/samvera/branding/5cd101a4-e698-4141-8c02-8976e14efe49/thumbnail/eddie-baby.jpg",
```

this should remove app/samvera

**Comment by jillpe** [comment]
SoftServ QA:
On [DEMO](https://demo.hyku-demo.notch8.cloud/), Login as admin
Create a New User Collection
After clicking save, the edit page should have a "Branding" tab. Click it, and notice the "Thumbnail" section at the bottom of the form
Upload an image and save. Note, Alt Text may look out of place - this is a known bug and considered a separate issue. Alt text is optional - enter it if you'd like.
Click Collections in the sidebar to view all the collections (index page)
The thumbnail should appear and match the file you just uploaded
If you entered alt text, you should see if when you inspect the thumbnail element of that page. ie: see eddie-baby-alt-text

<details><summary<>'Thumbnail' section is present in the branding tab</summary>

![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/4e854312-99a3-4a25-af0f-f3a0b5fd7027)
<details/>

[Collection Created](https://demo.hyku-demo.notch8.cloud/dashboard/collections/6e354675-b509-44bf-a7f7-dfcf0fa23012?locale=en)
<details><summary>thumbnail displays but the alt text is not present in the dashboard collection page view, public collection show page view, or the catalog search view. The catalog search results thumbnail also overlaps text</summary>


![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/be00474e-2c28-43d1-af05-884e1de1d3e3)



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/98e35505-9061-448b-80c0-74018855c411)




![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/62d79958-fa14-42b1-8132-c2313e29c069)


</details>

**Comment by ShanaLMoore** [comment]
![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/10081604/3128a264-6e1e-43e9-9634-ff0fa1203632)

**Comment by jillpe** [comment]
Blocked until knapsack is updated

**Comment by jillpe** [comment]
SoftServ QA: ✅ 



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/534d5586-0af0-47ba-9a0e-980b062a0684)



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/00b71bdc-5b8b-4190-a07c-319cd74e556a)



![Image](https://github.com/scientist-softserv/hykuup_knapsack/assets/84697174/65875f63-4869-485c-9895-0a065d2c5714)

**Comment by ShanaLMoore** [comment]
issues noticed internally when QA-ing collection branding. client verification is not needed.
